### DISCUSSION TITLE
Search on a relationship column

### DISCUSSION THREAD

**Discussion by johanboekhoven** [aminalaee/sqladmin#568] [General]
I've subclassed ModelView and I'm overwrting  def search_query(self, stmt: Select, term: str) -> Select:
This all works wonderfully well on normal fields, but now I would like to add a custom term that will look into the relationships.

Say I have a task relationship with the status set to finished, I'd like to know if I can make a statement that would work for this situation.
I tried many things, but can't get something to work. I know the statement sqlalchemy statement is wrapped in in the main search statement, so maybe this is not possible at all? My sqlalchemy skills are failing me.

I need something along the lines of this pseudo code:
func.count(select(models.Task).where(models.Task.task_status == task_status.FINISHED))==2
So I want to find out if the relationship Task items have their status set to finished and that there are 2 items that have done so.

Thanks for any info!

**Comment by aminalaee** [comment]
So are you searching in the Task page? Or the relationship page?
Can you post your example model?

**Reply by johanboekhoven** [reply]
I'm searching from the Asset page, the asset model has the task relationship.
I've just left for an extended period, so don't have the code at hand unfortunately and I'm not in a rush. Thanks for the quick response!
